src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -930,7 +930,7 @@ public override object VisitField(FieldSymbol symbol, TypeCompilationState argum
                     var lazySemanticModel = body == null ? null : new Lazy<SemanticModel>(() =>
                     {
                         var syntax = body.Syntax;
-                        var semanticModel = (CSharpSemanticModel)_compilation.GetSemanticModel(syntax.SyntaxTree);
+                        var semanticModel = (CSharpSemanticModel)AnalyzerDriver.GetOrCreateCachedSemanticModel(syntax.SyntaxTree, _compilation, _cancellationToken);
```

Comment:
Two problems here. First, this appears to be crossing abstraction boundaries. AnalyzerDriver is logically in a separate API layer built over the compiler. Second, a consequence of this chance would be to cause garbage to be retained unnecessarily long. The creation of a fresh semantic model was intentional.